next/tp6/env_pendulum.py

```diff
@@ -1,142 +1,186 @@
-    def step(self,u):
-        x,c=env_abstract.EnvDiscretized.step(self,u)
+
+    def step(self, u):
+        x, c = env_abstract.EnvDiscretized.step(self, u)
         if self.withSimpleCost:
-            c = int(np.all(np.abs(self.conti.x)<1e-3))
-        return x,c
+            c = int(np.all(np.abs(self.conti.x) < 1e-3))
+        return x, c
+
     def jupyter_cell(self):
         return self.conti.viewer.viewer.jupyter_cell()
-        
+
+
 class EnvPendulumSinCos(env_abstract.EnvPartiallyObservable):
-    def __init__(self,nbJoint=1,**kwargs):
-        env = EnvPendulum(nbJoint,**kwargs)
-        def sincos(x,nq):
-            q,v = x[:nq],x[nq:]
-            return np.concatenate([np.concatenate([(np.cos(qi),np.sin(qi)) for qi in q]),v])
-        def atan(x,nq):
-            cq,sq,v = x[:2*nq:2],x[1:2*nq:2],x[2*nq:]
-            return np.concatenate([np.arctan2(sq,cq),v])
-        env_abstract.EnvPartiallyObservable.__init__(self,env,nobs=env.nq*2+env.nv,
-                                                     obs=lambda x:sincos(x,env.nq),
-                                                     obsInv=lambda csv:atan(csv,env.nq))
+    def __init__(self, nbJoint=1, **kwargs):
+        env = EnvPendulum(nbJoint, **kwargs)
+
+        def sincos(x, nq):
+            q, v = x[:nq], x[nq:]
+            return np.concatenate(
+                [np.concatenate([(np.cos(qi), np.sin(qi)) for qi in q]), v]
+            )
+
+        def atan(x, nq):
+            cq, sq, v = x[: 2 * nq : 2], x[1 : 2 * nq : 2], x[2 * nq :]
+            return np.concatenate([np.arctan2(sq, cq), v])
+
+        env_abstract.EnvPartiallyObservable.__init__(
+            self,
+            env,
+            nobs=env.nq * 2 + env.nv,
+            obs=lambda x: sincos(x, env.nq),
+            obsInv=lambda csv: atan(csv, env.nq),
+        )
         self.reset()
+
     def jupyter_cell(self):
         return self.full.viewer.jupyter_cell()
-        
+
+
 class EnvPendulumHybrid(env_abstract.EnvDiscretized):
-    def __init__(self,nbJoint=1,**kwargs):
-        env = EnvPendulumSinCos(nbJoint,**kwargs)
-        NU = 21 #11
-        env_abstract.EnvDiscretized.__init__(self,env,discretize_x=0,discretize_u=NU)
-        self.conti.full.Kf=.1   # Reduced friction, because larger steps would make friction unstable.
-        self.conti.full.DT=1e-1 # 5e-1 # Larger integration step to allow larger discretization grid
-        self.conti.full.costWeights = {'q': 0, 'tip': 1.0, 'u': 0.00, 'v': 0.1 }
+    def __init__(self, nbJoint=1, **kwargs):
+        env = EnvPendulumSinCos(nbJoint, **kwargs)
+        NU = 21  # 11
+        env_abstract.EnvDiscretized.__init__(self, env, discretize_x=0, discretize_u=NU)
+        self.conti.full.Kf = (
+            0.1  # Reduced friction, because larger steps would make friction unstable.
+        )
+        self.conti.full.DT = (
+            1e-1  # 5e-1 # Larger integration step to allow larger discretization grid
+        )
```

Comment:
same here: if this doesn't fit on one line, we can put the comment on its own line, above.